In [1]:
import numpy as np
import pandas as pd
import requests
import json

In [2]:
subreddit = pd.read_csv('./workfile.csv')

In [3]:
subreddit.sub_name.values[25]

'yankees'

In [4]:
df_sub = pd.read_csv('./data/{}_final.csv'.format(subreddit.sub_name.values[25]))

In [5]:
df_sub.head()

Unnamed: 0               author                      comments_author_lst  \
0           0  imonlyhereforsports                                       []   
1           1           blu3sclu3s  ['PorzingisIsGod', 'BanterCityPodcast']   
2           2     westonroadflowss                                       []   
3           3                krese                                       []   
4           4        NYRangers1313                        ['Sun-Devil4646']   

  comments_score_lst   created_utc  guilded  num_comments  pinned    r_id  \
0                 []  1.538261e+09        0             4   False  9k179b   
1            [18, 8]  1.538155e+09        0            25   False  9jouru   
2                 []  1.538186e+09        0             1   False  9jsvy1   
3                 []  1.538181e+09        0             1   False  9jsb7l   
4                [8]  1.538090e+09        0             3   False  9jhsjz   

   score subreddit                                              title  
0     31   yankees                            Thumbs down guy spotted  
1    160   yankees  Shirt I whipped up to wear to the Wild Card ga...  
2      2   yankees  Why does Michael keep getting pissed about the...  
3      1   yankees  boston should play hard this weekend with best...  
4     61   yankees  My shitpost of the day. My CC Sabathia Banepos...

In [6]:
df_sub.shape

(4800, 12)

In [7]:
df_sub.drop_duplicates('author',inplace=True)

In [8]:
auth_lst=df_sub.author.values

In [9]:
len(auth_lst)

242

- AskOuija start 4800 end 1580
- Futurology start 4800 end 126
- GameLive_0n start 4800 end 1
- Jokes start 4800 end 875
- NBA2k start 4800 end 961
- NoStupidQuestions start 4800 end 1801
- PoliticalHumor start 4800 end 638
- Showerthoughts start 4800 end 1881
- TwoXChromosomes start 4800 end 534
- europe starst 4800 end 393
- explainlikeimfive starts 4800 end 522
- funny start 4800 end 1525
- hiphopheads start 4800 end 702
- india start 4800 end 444
- lakers start 4800 end 264
- news start 4800 end 226
- nfl start 4800 end 443
- nottheonion start 4800 end 225
- opieandanthony start 4800 end 383
- science start 4800 end 177
- soccer start 4800 end 341
- technology start 4800 end 251
- timberwolves start 4800 end 270
- todayilearned start 4800 end 536
- videos start 4800 end 946
- yankees start 4800 end 242

In [10]:
sub_lst = []
for i in auth_lst:
    json_link  = 'https://www.reddit.com/user/{}/.json'.format(i)
    res = requests.get(json_link, headers={'User-agent': 'Capstone Bot 0.1'})
    data = res.json()
    try:
        df=pd.DataFrame(data)
        try:
            df=pd.DataFrame(df.T.children[1])
        except:
            continue
        for i in range(24):
            if len(df.data[i])==92:
                sub_lst.append(df.data[i]['subreddit'])
        print(sub_lst[-1])
    except: 
        continue
        
    

rangers
k12sysadmin
k12sysadmin
k12sysadmin
baseball
NYYankees
sto
yankees
OSHA
GreenBayPackers
DisneyWorldResorts
DisneyWorldResorts
AskReddit
AskReddit
MaddenUltimateTeam
travisscott
Blackops4
NYYankees
yankees
yeezys
yeezys
yankees
movies
Rebelution
yankees
yankees
yankees
yankees
diabetes_t1
KidsAreFuckingStupid
baseball
The_Donald
AskOuija
listentothis
AMCsAList
Kanye
fantasyfootball
yankees
yankees
lgg6
dankmemes
oakland
investing
yankees
yankees
brockhampton
videos
juul
Overwatch
yankees
yankees
nba
nba
yankees
WWEChampions
WWEChampions
nyjets
Neuropsychology
reactiongifs
reactiongifs
reactiongifs
Denver
Denver
yankees
yankees
grandpajoehate
Broadway
AskReddit
AskReddit
BillyJoel
whatisthisthing
whatisthisthing
AskReddit
AskReddit
CharlotteHornets
FIFA
yankees
yankees
Fitness
redsox
lockn
Advice
yankees
yankees
nycrail
nycrail
yankees
gaming
fantasybball
marketing
CompTIA
sports
PuertoRico
BBQ
BBQ
EDM
EDM
yankees
nba
news
pokemon
SeaJobs
Cooking
Cooking
yankees
yankees
yankees
7

In [11]:
df = pd.DataFrame({'subreddits':sub_lst})

In [12]:
df.to_csv('./data/{}_posters_lst.csv'.format(subreddit.sub_name.values[25]),index=False)

In [13]:
#json_link  = 'https://www.reddit.com/user/YoungSandwich18/.json'

In [14]:
#res = requests.get(json_link, headers={'User-agent': 'Capstone Bot 0.1'})

In [15]:
#data = res.json()

In [16]:
#df=pd.DataFrame(data)

In [17]:
#df.T.children[1]

In [18]:
#df = pd.DataFrame(df.T.children[1])

In [19]:
#df.data[0]['subreddit']

In [20]:
#len(df.data[0])

In [21]:
#len(df.data[25]) #max is 24